ci/check-crates.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# env:
+#   - CRATE_TOKEN
+
+declare skip_patterns=(
+    "./Cargo.toml"
+    "./programs/sbf/Cargo.toml"
+)
+
+declare -A verified_crate_owners=(
+    ["solana-grimes"]=1
+)
+
+# search all Cargo.toml
+readarray -t files <<<"$(find . -name "Cargo.toml")"
+
+declare -A including_unverified_owners=()
+declare -A unknown_errors=()
+
+for file in "${files[@]}"; do
+    for skip_pattern in "${skip_patterns[@]}"; do
+        if [[ $file =~ ^$skip_pattern ]]; then
+            continue 2
+        fi
+    done
+
+    crate_name=$(grep -Poz '[\s\S]*?package\][\s\S]*?name = "\K([a-zA-Z0-9_\-]*)' "$file" | tr -d '\0')
+    echo "=== $crate_name ($file) ==="
+
+    readarray -t owners <<<"$(cargo owner --list -q "$crate_name" --token "$CRATE_TOKEN" 2>&1)"
+
+    for owner in "${owners[@]}"; do
+        if [[ -z $owner ]]; then
+            continue
+        fi
+        if [[ $owner =~ ^error ]]; then
+            unknown_errors+=(["$crate_name ($file)"]=1)
+            echo "❌ ERROR"
```

Comment:
i think we can filter out the errors that are due to being unpublished and provide a more informative message.  last line contains "(status 404 Not Found)"

Reply:
oh yeah. it is just a sign to make us find the error easily haha. actually it will print something like this:
<img width="685" alt="Screenshot 2022-12-02 at 12 37 59 PM" src="https://user-images.githubusercontent.com/8209234/205215864-f5b95f85-112e-4f23-9250-91718606695e.png">

or you mean we only print the lats line 🤔

Reply:
meant to prompt the developer with something more helpful. maybe something like
```
new crate $CRATE_NAME not found on crates.io. make a dummy publication (hint to
instructions or mark it as not for publication with `package.publish = false`
in its Cargo.toml
```